### PR TITLE
fix:regional samples for SM

### PR DESCRIPTION
**Pull request by vipul7499**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [ ] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: secretmanager/snippets/regional_samples/snippets_test.py [hunk 1]
```diff
@@ -286,6 +292,28 @@ def test_get_regional_secret_version(
     assert secret_id in version.name
     assert version_id in version.name
 
+def test_iam_grant_access_with_regional_secret(
+    regional_client: secretmanager_v1.SecretManagerServiceClient,
+    regional_secret: Tuple[str, str, str, str],
```
Comment (msampathkumar) [inline secretmanager/snippets/regional_samples/snippets_test.py:297]: This is not a good practice. Kindly find a better alternative.

```
    regional_secret: Tuple[str, str, str, str],
....
....
    project_id, location_id, secret_id, _ = regional_secret
```

Comment (vipul7499) [reply secretmanager/snippets/regional_samples/snippets_test.py:297]: Can you suggest an alternative? 
The only alternative i can think of is to decompose the list into each variable which will make the code messy. 
Also, path of the secret is project/{project_id}/location/{location_id}/secret/{secret_id}.

Comment (msampathkumar) [reply secretmanager/snippets/regional_samples/snippets_test.py:297]: Sure. A few suggestions on your overall code

1. Generally following CLI code is not encouraged. Your code can be simpler.
```
    parser = argparse.ArgumentParser(
        description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter
    )
    parser.add_argument("project_id", help="id of the GCP project")
    parser.add_argument("location_id", help="id of location where secret is stored")
```

2. Do decompose the variables. Messy code is better complicated.

    (Better alternative) In test files, `project_id` & `location` could be global variables. Ex: https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/generative_ai/context_caching/context_caching_test.py#L26-L28

Comment (vipul7499) [reply secretmanager/snippets/regional_samples/snippets_test.py]: I have fixed 2. Please check.

For 1, it is not included in the samples, it is just there for someone, if they want to run a single sample.

**Review by msampathkumar — APPROVED** [review]
LGTM

**Pull request merged**